config/mbed/chip-gn/args.gni

```diff
@@ -0,0 +1,39 @@
+# Copyright (c) 2020 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build_overrides/chip.gni")
+
+chip_device_platform = "mbed"
+
+chip_project_config_include = ""
+chip_system_project_config_include = ""
+chip_device_project_config_include = ""
+
+chip_inet_config_enable_raw_endpoint = false
+chip_inet_config_enable_udp_endpoint = true
+chip_inet_config_enable_tcp_endpoint = true
+chip_inet_config_enable_dns_resolver = true
+chip_inet_config_enable_async_dns_sockets = false
+
+chip_custom_build_cflags = [
+  "-Wno-shadow",
+  "-Wno-conversion",
+  "-Wno-maybe-uninitialized",
+  "-Wno-return-type",
+]
+
+custom_toolchain = "//toolchain:mbed"
+mbedtls_target = "//mbedtls:mbedtls"
+
+treat_warnings_as_errors = false
```

Comment:
Is it necessary? I think if the problem is with mbedOS headers they can be included using `-isystem` instead `-I` and the rest should compile without warnings, right?

Reply:
> #### Problem
> There is no support for Mbed OS in CHIP.
> 
> #### Summary of Changes
> Add Mbed OS support to CHIP.
> 
> It requires the PR #6352 to be merged and the Mbed OS docker image to be pushed on dockerhub to pass the build workflow.
> 
> Few details about the port:
> 
> * Our main development target is the [PSOC6 WiFi-BT Prototyping kit](https://os.mbed.com/platforms/CY8CPROTO-062-4343W/) from cypress:
> * The port only support WiFi and BLE connectivity; not Thread.
> * There is only two examples ported at the moment: Shell and the lock application. More will come as we continue our effort on CHIP.
> * There is good integration with VSCode, build, flash and debug of the examples supported is provided.
> 
> There is few commits in this PR which could be their own PRs. I want guidance on what I should do with them:
> 
> * [503f4a1](https://github.com/project-chip/connectedhomeip/commit/503f4a1eba4c4f4d9ec2457058e90ef53d6469f8): Prevent mutex initialization if locking is not supported.
> * [a4d4181](https://github.com/project-chip/connectedhomeip/commit/a4d4181fc69f02d7ecc9f1ada4551b1c01ceb703): Avoid use of -Wno-unknown-warning-option in GCC.
> * ~[ec3566b](https://github.com/project-chip/connectedhomeip/commit/ec3566b4de1cfc1951afcc157bd89af8e8f71262): Update the Android workflow to archive the APK created by the build.~
> * [b693df4](https://github.com/project-chip/connectedhomeip/commit/b693df4bb6f03269c3058a0d926e875122c30d39): Small fix to include C99 format macro constant in some files using them.
> * [726ded3](https://github.com/project-chip/connectedhomeip/commit/726ded34076f7c7ddf03d33937c1f47e04f921e0): Compilation fix for LwIP 2.
> 
> In term of future work here is what the Mbed OS team supporting CHIP will work on next:
> 
> * Unit test support: We're adding Mbed OS support in the `test_driver`
> * Documentation: This submission is scarce in term of documentation for Mbed OS users we would like to provide a guide that explain how to build a CHIP application with Mbed OS.
> * Example support: We want to support more examples starting with the lightning app then more will follow.
> * Integration testing: It isn't great that no on-device test is run. We've read [CHIP on-device testing](https://github.com/project-chip/connectedhomeip/blob/master/docs/api/device_runner.md) document but it seems there isn't a lot of movement around it at the moment. We would like to contribute in that area.

@rgoliver

Reply:
Fixed, when the CHIP library is compiled `-isystem` is used. 
Other warnings have been removed (sometime with Pragma)